### DISCUSSION TITLE
黑道圣徒4 Saints Row IV 无法启动

### DISCUSSION THREAD

**Discussion by izumi-sora** [Gamuxorg/bbs#330] [游戏或网站运行反馈]
### 以下是推荐模板
游戏名称：黑道圣徒4 Saints Row IV

操作系统：*（uname -a）*
Operating System: Debian GNU/Linux 11
KDE Plasma Version: 5.20.5
KDE Frameworks Version: 5.78.0
Qt Version: 5.15.2
Kernel Version: 5.10.0-10-amd64
OS Type: 64-bit
Processors: 4 × Intel® Core™ i5-7300HQ CPU @ 2.50GHz
Memory: 7.7 GiB of RAM
Graphics Processor: Mesa Intel® HD Graphics 630

bug描述：*（具体说说发生了什么）*首先是没可执行权限，然后我给了
```
chmod +x *
 ```
然后虽然有终端就输出了下面的话，

终端运行输出 ：*（如果有的话）*就是这点，没其他的了
./saferun.sh: 21: ./SaintsRow4.i386: not found
        不是动态可执行文件
./SaintsRow4.i386 failed to start due to the following dependancy problems:

自行尝试解决bug的方式和结果：*（如果有的话）*
准备安装一些lib*i386软件试试

**Comment by izumi-sora** [comment]
刚安装libc6-i386就有效果了
终端输出：
./SaintsRow4.i386: error while loading shared libraries: libX11.so.6: cannot open shared object file: No such file or directory
./SaintsRow4.i386 failed to start due to the following dependancy problems:
        linux-gate.so.1 (0xf7f1a000)
        libX11.so.6 => not found
        libGL.so.1 => not found
        libSDL2-2.0.so.0 => ./libSDL2-2.0.so.0 (0xf7de7000)
        librt.so.1 => /lib32/librt.so.1 (0xf7ddc000)
        libdl.so.2 => /lib32/libdl.so.2 (0xf7dd6000)
        libsteam_api.so => ./libsteam_api.so (0xf7d2a000)
        libfreetype.so.6 => not found
        libopenal-eon.so.1 => ./libopenal-eon.so.1 (0xf7cc0000)
        libz.so.1 => not found
        libstdc++.so.6 => not found
        libm.so.6 => /lib32/libm.so.6 (0xf7bba000)
        libgcc_s.so.1 => not found
        libpthread.so.0 => /lib32/libpthread.so.0 (0xf7b98000)
        libc.so.6 => /lib32/libc.so.6 (0xf79af000)
        libasound.so.2 => not found
        libpulse-simple.so.0 => not found
        libpulse.so.0 => not found
        libX11.so.6 => not found
        libXext.so.6 => not found
        libXcursor.so.1 => not found
        libXinerama.so.1 => not found
        libXi.so.6 => not found
        libXrandr.so.2 => not found
        libXss.so.1 => ./libXss.so.1 (0xf79a9000)
        /lib/ld-linux.so.2 (0xf7f1c000)
        libXxf86vm.so.1 => not found
        libwayland-egl.so.1 => not found
        libwayland-client.so.0 => not found
        libwayland-cursor.so.0 => not found
        libxkbcommon.so.0 => not found
        libstdc++.so.6 => not found
        libX11.so.6 => not found
        libXext.so.6 => not found

**Comment by hyh19962008** [comment]
把各种缺失的32位依赖安装上（显示not found的）：
sudo apt install libgl1-mesa-dri:i386 libx11-6:i386

**Comment by izumi-sora** [comment]
抱歉，装不了，缩减之后搜到的都安装了
sudo apt install libgl1-mesa-dri:i386 libx11-6:i386
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libgl1-mesa-dri:i386
E: Unable to locate package libx11-6:i386

**Comment by izumi-sora** [comment]
> 把各种缺失的32位依赖安装上（显示not found的）： sudo apt install libgl1-mesa-dri:i386 libx11-6:i386

我安装闭源驱动试试

**Comment by izumi-sora** [comment]
整不了，换ubuntu了，

**Comment by izumi-sora** [comment]
找到办法了，跟hyh19962008说的一样，只不过如果libgl1-mesa-dri:i386没找到，就sudo apt search libgl1-mesa-dri，然后就能看到适合软件包，只是后面必须有:i386

**Comment by izumi-sora** [comment]
举个例子
sudo apt search libxcursor
p   libxcursor-dev                                                                                                 - X cursor management library (development files)                                                                          
p   libxcursor-dev:i386                                                                                            - X cursor management library (development files)                                                                          
i   libxcursor1                                                                                                    - X cursor management library                                                                                              
i   libxcursor1:i386                                                                                               - X cursor management library    

就可以选择安装libxcursor1:i386
